### PR TITLE
adding network policy docs

### PR DESCRIPTION
**Pull request by puja108**
Explaining Ingress and Egress Network Policies.

Egress based on tests in https://github.com/giantswarm/adidas/issues/96#issuecomment-320284881

### CODE REVIEW

**Comment by puja108** [comment]
RFR

File: content/guides/limiting-pod-communication-with-network-policies/index.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      selector: calico/k8s_ns == 'trusted'
+```
+
+and apply the policy with
+
+```bash
+calicoctl create -f trusted-namespace.yaml
+```
+
+Now connections to `8.8.8.8` will only be possible from Pods living in the namespace called `trusted`.
```
Comment (webwurst) [inline content/guides/limiting-pod-communication-with-network-policies/index.md]: > Now Pods living in the namespace called `trusted` are only possible to connect to `8.8.8.8` and nowhere else.

Should it be like that? If I understood correctly.

File: content/guides/limiting-pod-communication-with-network-policies/index.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
++++
+title = "Limiting Pod Communication with Network Policies"
+description = "Guide on how to limit Pod communication using Network Policies"
+date = "2017-08-08"
+type = "page"
+weight = 30
```
Comment (marians) [inline content/guides/limiting-pod-communication-with-network-policies/index.md]: Maybe make this a 60? Not as important as persistent volumes, for example, IMO.

File: content/guides/limiting-pod-communication-with-network-policies/index.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+You can limit communication to Pods using the Network Policy API of Kubernetes.
+
+Currently the API only includes Ingress rules, i.e. you can only limit traffic to groups of Pods. Egress rules, limiting outgoing traffic from Pods, are coming to the API soon. Until then you have to use the Egress functionality of our Network Plugin, Calico, directly.
+
+Here we give an overview and introduction of how to create and use these policies.
+
+## Ingress Network Policies
+
+`NetworkPolicy` resources in Kubernetes use labels to select pods and define rules which specify what traffic is allowed to the selected pods.
```
Comment (marians) [inline content/guides/limiting-pod-communication-with-network-policies/index.md]: For consistency reasons, I'd remove the backticks around the NetworkPolicy resource name.

File: content/guides/limiting-pod-communication-with-network-policies/index.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+apiVersion: extensions/v1beta1
+metadata:
+  name: default-deny
+spec:
+  podSelector:
+```
+
+Applying the above policy with 
+
+```bash
```
Comment (marians) [inline content/guides/limiting-pod-communication-with-network-policies/index.md]: As bash highlighting is pretty erratic, I use `nohighlight` in this case.

File: content/guides/limiting-pod-communication-with-network-policies/index.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+            matchLabels:
+              role: frontend
+      ports:
+        - protocol: TCP
+          port: 6379
+```
+
+You need to apply this policy to the Namespace that the backend Pods live in.
+
+```bash
```
Comment (marians) [inline content/guides/limiting-pod-communication-with-network-policies/index.md]: nohighlight

File: content/guides/limiting-pod-communication-with-network-policies/index.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+  egress:
+  - action: deny
+    destination:
+      net: 8.8.8.8
+    source: {}
+```
+
+We apply the policy with
+
+```bash
```
Comment (marians) [inline content/guides/limiting-pod-communication-with-network-policies/index.md]: nohighlight

File: content/guides/limiting-pod-communication-with-network-policies/index.md (hunk 1 above)
Comment (puja108) [inline content/guides/limiting-pod-communication-with-network-policies/index.md]: Right, I will change the text to make that more clear.

Comment (marians) [inline content/guides/limiting-pod-communication-with-network-policies/index.md]: I understood it that way, yes.

**Comment by puja108** [comment]
I'll be adding another section about rules for pods that need to serve outside traffic (ingress).

**Pull request merged**